lib/chef/provider/package/zypper.rb

```diff
@@ -70,7 +71,34 @@ def get_current_versions
         end
 
         def candidate_version
-          @candidate_version ||= package_name_array.each_with_index.map { |pkg, i| available_version(i) }
+          if source_files_exist?
+            unless uri_scheme?(new_resource.source) || ::File.exist?(new_resource.source)
```

Comment:
we shouldn't actually ever get here with a uri, because `source_files_exist?` would be false.